File: project/server/src/main/java/ru/zagrebin/dto/SubscriptionDto.java
package ru.zagrebin.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SubscriptionDto {
    private Long userId;
    private boolean subscribed;
    private int followersCount;
    private int followingCount;
}

File: project/mobile/app/src/main/java/ru/zagrebin/culinaryblog/data/remote/dto/SubscriptionDto.kt
package ru.zagrebin.culinaryblog.data.remote.dto

import ru.zagrebin.culinaryblog.model.SubscriptionStatus

data class SubscriptionDto(
    val userId: Long,
    val subscribed: Boolean,
    val followersCount: Int,
    val followingCount: Int
)

fun SubscriptionDto.toModel(): SubscriptionStatus = SubscriptionStatus(
    userId = userId,
    subscribed = subscribed,
    followersCount = followersCount,
    followingCount = followingCount
)
